buildconfig/CMake/CommonSetup.cmake

```diff
@@ -412,6 +402,30 @@ if(NOT BUNDLES)
   set(BUNDLES "./")
 endif()
 
+# ##############################################################################
+# Setup pre-commit here as otherwise it will be overwritten by earlier
+# pre-commit hooks being added
+# ##############################################################################
+option(ENABLE_PRECOMMIT "Enable pre-commit framework" ON)
+if (ENABLE_PRECOMMIT)
+  if (MSVC)
+    # Use downloaded ThirdParty version of pre-commit
+    execute_process(COMMAND bash -c "${MSVC_PYTHON_EXECUTABLE_DIR}/Scripts/pre-commit.cmd install" WORKING_DIRECTORY ${PROJECT_SOURCE_DIR})
+    # Create pre-commit script wrapper to use mantid third party python for pre-commit
```

Comment:
I think it would be good to replace these next 2 `execute_process` commands and use [`file`](https://cmake.org/cmake/help/v3.15/command/file.html?highlight=file) instead to avoid starting 2 new subprocesses that are particularly expensive on windows.